src/groupeddatatable/grouping.jl

```diff
@@ -371,12 +369,11 @@ end
 
 function _makeheaders{T<:Function}(fs::Vector{T}, cn::Vector{Symbol})
     fnames = _fnames(fs) # see other/utils.jl
-    reshape([Symbol(colname,'_',fname) for fname in fnames, colname in cn],
-            length(fnames)*length(cn))
+    [Symbol(colname,'_',fname) for fname in fnames for colname in cn]
 end
 
 function _aggregate{T<:Function}(d::AbstractDataTable, fs::Vector{T}, headers::Vector{Symbol}, sort::Bool=false)
-    res = DataTable(colwise(fs, d), headers)
+    res = DataTable(Any[vcat(f(d[i])) for f in fs for i in 1:size(d, 2)], headers)
```

Comment:
Hadn't seen you comment. What's the advantage of wrapping `1` inside `[1]`? With something like https://github.com/JuliaStats/DataFrames.jl/pull/882, a scalar would be recycled, which could be more useful than a 1-element array.